lib/octocatalog-diff/catalog-util/builddir.rb

```diff
@@ -330,7 +334,7 @@ def install_ssl_client_password(logger, password)
       end
 
       def environment
-        @options[:preserve_environments] ? @options.fetch(:environment, 'production') : 'production'
+        @options.fetch(:environment, 'production')
```

Comment:
I looked into this and you're actually very close here. There is a method in https://github.com/github/octocatalog-diff/blob/master/lib/octocatalog-diff/catalog/computed.rb#L82-L84 that needs to be updated to match this change. Also some spec tests need to be updated for the new behavior.

Have a look at https://gist.github.com/kpaulisse/7bda51e395aab49585aa9933d78fe724 for what I had to do to get the tests passing. You can probably remove the warning about `--environment` having no effect, since it does now. Anyway, hopefully this is a push in the right direction - I'm 👍 to the change in general.